### PR TITLE
[0.12] Update release-notes.md

### PR DESCRIPTION
**Pull request by maflcko**
- Transaction memory pool limiting
- Priority transactions
- Wallet transaction fees

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Pending 995b9f385b935e4e9b9fa46e82f642204cc85cba to get merged. Also, I am not a native speaker, so feedback on syntax or content is appreciated.

**Comment by jonasschnelli** [comment]
utACK (ut=english looks good but i'm not a lector)

> Wallet transaction fees

Should we mention the tiny API change in 0.12 when using settxfee and the GUI?

Still missing:
* ban functions (setban / banlist / ban persistence)
* wallet can be used with pruning
* ...

File: doc/release-notes.md [hunk 1]
```diff
@@ -215,6 +215,65 @@ of just announcing the hash. In a reorganization, all new headers are sent,
...
+Core to use floating fees. By default, Bitcoin Core will use floating
+fees.
+
+Based on past transaction data, floating fees approximate the fees
+required to get into the `m`th block from now. This is configurable
+with `-txconfirmtarget=<m>` (default: `2`).
+
+Sometimes, it is not possible to give good estimates, or an estimate
+at all. Therefore, a fallback value can be set with `-fallbackfee=<f>`
+(default: `FIXME`).
```
Comment (MarcoFalke) [inline doc/release-notes.md]: This would require e420a1b (#7296)

File: doc/release-notes.md [hunk 2]
```diff
@@ -215,6 +215,65 @@ of just announcing the hash. In a reorganization, all new headers are sent,
...
+
+At all times, Bitcoin Core will cap fees at `-maxtxfee=<x>` (default:
+0.10) BTC.
```
Comment (MarcoFalke) [inline doc/release-notes.md]: Related to  #7084

File: doc/release-notes.md [hunk 3]
```diff
@@ -215,6 +215,65 @@ of just announcing the hash. In a reorganization, all new headers are sent,
...
+Furthermore, Bitcoin Core will never create transactions smaller than
+the current minimum relay fee.
+Finally, a user can set the minimum fee rate for all transactions with
+`-mintxfee=<i>`, which defaults to 1000 satoshis per kB.
```
Comment (MarcoFalke) [inline doc/release-notes.md]: This would require 995b9f3 (part of #7296 and imo a mandatory backport for 0.12)

File: doc/release-notes.md [hunk 4]
```diff
@@ -215,6 +215,65 @@ of just announcing the hash. In a reorganization, all new headers are sent,
 instead of just the new tip. This can often prevent an extra roundtrip before
 the actual block is downloaded.
 
+Memory pool limiting
+--------------------------------
+
+Previous versions of Bitcoin Core had their mempool limited by checking
+a transaction's fees against the node's minimum relay fee. There was no
+upper bound on the size of the mempool and attackers could send massive
+amounts of transactions paying just slighly more than the default minimum
```
Comment (fanquake) [inline doc/release-notes.md]: s/massive amounts of/a large number of

File: doc/release-notes.md [hunk 5]
```diff
@@ -215,6 +215,65 @@ of just announcing the hash. In a reorganization, all new headers are sent,
...
+it's maximum size, the transaction with the lowest feerate will be
+evicted and the node's minimum relay fee will be increased to match
+this feerate. The initial minimum relay fee is set to 1000 satoshis
+per kB.
+
+Priority transactions
+---------------------
+
+Transactions that do not pay the minimum relay fee, are called "free
+transactions" or priority transactions. Previous versions of Bitcoin
```
Comment (fanquake) [inline doc/release-notes.md]: Should priority transactions be in quotes as well?

File: doc/release-notes.md [hunk 6]
```diff
@@ -215,6 +215,65 @@ of just announcing the hash. In a reorganization, all new headers are sent,
...
+Core would relay and mine priority transactions depending on their
+setting of `-limitfreerelay=<r>` (default: `r=15` kB per minute) and
+`-blockprioritysize=<s>` (default: `50000` bytes of a block's
+priority space).
+
+Priority code is planned to get moved out of from Bitcoin Core 0.13
+and the default block priority size was set to `0` in Bitcoin Core
```
Comment (fanquake) [inline doc/release-notes.md]: s/was/has been

File: doc/release-notes.md [hunk 7]
```diff
@@ -215,6 +215,65 @@ of just announcing the hash. In a reorganization, all new headers are sent,
...
+0.12.
+
+Wallet transaction fees
+-----------------------
+
+Various impromements were made how the wallet calculates transaction
```
Comment (fanquake) [inline doc/release-notes.md]: s/impromements were/improvements have been made to

File: doc/release-notes.md (hunk 5 above)
Comment (MarcoFalke) [inline doc/release-notes.md]: I think this is the "official" term.

File: doc/release-notes.md (hunk 6 above)
Comment (MarcoFalke) [inline doc/release-notes.md]: Thx!

File: doc/release-notes.md [hunk 8]
```diff
@@ -215,6 +226,65 @@ of just announcing the hash. In a reorganization, all new headers are sent,
...
+upper bound on the size of the mempool and attackers could send a large
+number of transactions paying just slighly more than the default minimum
+relay fee to crash nodes with relatively low RAM. A temporary workaround
+for previous versions of Bitcoin Core was to raise the default minimum
+relay fee.
+
+Bitcoin Core 0.12 will have a strict maximum size on the mempool. The
+default value is 300 MB and can be configured with the `-maxmempool`
+parameter. Whenever a transaction would cause the mempool to exceed
+it's maximum size, the transaction with the lowest feerate will be
```
Comment (dgenr8) [inline doc/release-notes.md]: its

**Comment by MarcoFalke** [comment]
> Still missing:
 *   ban functions (setban / banlist / ban persistence)
 *   wallet can be used with pruning
 *   ...

@jonasschnelli I think you are the best person to write one or two sentences about the ban functions. (I haven't looked into that at all)

**Pull request merged**